### PR TITLE
Custom validation

### PR DESCRIPTION
**Pull request by weili-govtech**
- this PR is to be reviewed after https://github.com/LifeSG/validation-schema-generator/pull/2
- added `addRule()` function to allow defining custom rules
- added custom rules
  - filled
  - empty
  - equals
  - notEquals
  - uinfin

### CODE REVIEW

**Review by qroll — CHANGES_REQUESTED** [review]
lgtm! just need additional tests for edge cases

File: src/__tests__/custom-rules/custom-rules.spec.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import "../../custom-rules";
+import { jsonToSchema } from "../../schema-generator";
+import { TestHelper } from "../../utils";
+
+const ERROR_MESSAGE = "test error message";
+
+describe("custom-rules", () => {
+	it.each`
```
[the remark this answers is not in the document]
Comment (weili-govtech) [reply src/__tests__/custom-rules/custom-rules.spec.ts:8]: I have added the unit tests for string and number, for arrays, the tests are added in the `fields` branch because there are no fields in branch that can do arrays: https://github.com/LifeSG/validation-schema-generator/blob/fields/src/__tests__/custom-rules/custom-rules.spec.ts

**Review by qroll — APPROVED** [review]
